code/game/objects/items/reagent_containers/hypospray.dm

```diff
@@ -166,6 +118,74 @@
 
 	return TRUE
 
+///Always draws reagents on right click
+/obj/item/reagent_containers/hypospray/afterattack_alternate(atom/A, mob/living/user)
+	if(!in_range(A, user) || !user.Adjacent(A)) //So we arent drawing reagent from a container behind a window
+		return FALSE
+	can_draw_reagent(A, user, TRUE)
+
+///If it's possible to draw from something. Will draw_blood() when targetting a carbon, or draw_reagent() when targetting a non-carbon
+///Returns TRUE if we successfully draw from something
+/obj/item/reagent_containers/hypospray/proc/can_draw_reagent(atom/A, mob/living/user, alternate_draw_mode)
+	if(!A.reagents)
+		return FALSE
+	if(!istype(user))
+		return FALSE
```

Comment:
This also needs to be in the attack proc(s), as its going to runtime otherwise on the range checks.